ui/src/layouts/BuildPublication.tsx

```diff
@@ -60,9 +60,29 @@ const BuildPublication: React.FC<BuildPublicationProps> = (props) => {
                     message: 'You have selected there is a conflict of interest, please provide a reason.'
                 };
             }
-            if (typeof store.conflictOfInterestStatus == 'undefined') {
+
+            if (typeof store.conflictOfInterestStatus === 'undefined') {
                 ready = { ready: false, message: 'You must select a conflict of interest option' };
             }
+
+            if (store.affiliationsStatus && !store.affiliations.length) {
+                ready = {
+                    ready: false,
+                    message: 'You have selected there are affiliations, please provide them'
+                };
+            }
+
+            if (!store.affiliationsStatus && !store.hasNoAffiliations) {
+                ready = {
+                    ready: false,
+                    message: 'You have selected there are affiliations, please provide them'
+                };
+            }
+
+            if (typeof store.affiliationsStatus === 'undefined') {
```

Comment:
this code block is obsolete. 'store.affiliationsStatus' will never be undefined with the current codebase